earthly/rust/Earthfile

```diff
@@ -73,18 +84,40 @@ RUST_SETUP:
         cargo +nightly --version
 
     # Install tools we use commonly with `cargo`.
-    RUN cargo install cargo-nextest --locked && \
-        cargo install cargo-chef --locked && \
-        cargo install kani-verifier --locked && \
-        cargo kani setup && \
-        cargo install refinery_cli --locked && \
-        cargo install cargo-machete --locked
+    IF [ "$with_cargo_nextest" = "true" ]
+        RUN cargo install cargo-nextest --locked
+    END
+    IF [ "$with_cargo_chef" = "true" ]
+        RUN cargo install cargo-chef --locked
+    END
+    IF [ "$with_kani" = "true" ]
+        RUN cargo install kani-verifier --locked && \
+        cargo kani setup
+    END
+    IF [ "$with_refinery" = "true" ]
+        RUN cargo install refinery_cli --locked
+    END
+    IF [ "$with_cargo_machete" = "true" ]
+        RUN cargo install cargo-machete --locked
+    END
```

Comment:
See above.  A tool either works everywhere or we don;t use it.

Reply:
You can eliminate `kani` and `chef`.
* `chef` just breaks builds, it was on my TODO to eliminate it anyway.
* `kani` either works everywhere or we don;t use it.  Just comment it our with a note saying. "can not be used because not supported on Mac M1+ devices"  So that next time someone suggests to use it we can see why we don't.

* refinery, nextest and machete should work everywhere and are not optional.